helix-core/src/movement.rs

```diff
@@ -48,35 +52,113 @@ pub fn move_horizontally(
     range.put_cursor(slice, new_pos, behaviour == Movement::Extend)
 }
 
+pub fn move_vertically_visual(
```

Comment:
I assume visual movement is slightly less efficient, can we use `move_vertically` internally if softwrap is disabled?

Reply:
Good point, done in cdac97be